### PR TITLE
merge word-lo-hi to main

### PR DESCRIPTION
**Pull request by hero78119**
### Description

This pr is to complete milestone `word-lo-hi` back to main branch.

Since `word-lo-hi` branch is out-of-sync with main branch due to few features merged to main recently, in order to save review effort to review 1 pr instead of 2 pr, a new branch `word-lo-hi-merge-main` is created which covers
- [x] based on latest word-lo-hi branch
- [x] remove `_word` suffix and remove deprecated table field
- [x] also sync main and resolve conflict
- [x] Fix problem on main branch: applied a workaround `max_rwc + 1 + 2` magic number mention in another issue https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1507
- [x] ~~investigate super circuit degree 9 -> 12~~ https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1499 After fixed, degree 9 -> 10

With this pr, reviewer just need to review once.

### Issue Link

https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1509

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Contents

to complete milestone `word-lo-hi`

### CODE REVIEW

**Comment by hero78119** [comment]
Updated: its ready for review for all ci checked passed

File: zkevm-circuits/src/super_circuit/test.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ fn super_circuit_degree() {
     SuperCircuit::configure_with_params(&mut cs, params);
     log::info!("super circuit degree: {}", cs.degree());
     log::info!("super circuit minimum_rows: {}", cs.minimum_rows());
-    assert!(cs.degree() <= 9);
+    assert!(cs.degree() <= 12);
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/super_circuit/test.rs:23]: Note: In this PR, the super_circuit degree is 12.

File: zkevm-circuits/src/evm_circuit/param.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ use std::collections::HashMap;
 // Step dimension
 pub(crate) const STEP_WIDTH: usize = 128;
 /// Step height
-pub const MAX_STEP_HEIGHT: usize = 21;
+pub const MAX_STEP_HEIGHT: usize = 19;
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/evm_circuit/param.rs:12]: Note: MAX_STEP_HEIGHT is reduced from 21 to 19.

File: bus-mapping/src/circuit_input_builder.rs [hunk 3]
```diff
@@ -374,7 +376,7 @@ impl CircuitInputBuilder<DynamicCParams> {
             // needed.
             let max_keccak_rows = 0;
             FixedCParams {
-                max_rws: max_rws + 1,
+                max_rws: max_rws + 3,
```
Comment (ChihChengLiang) [inline bus-mapping/src/circuit_input_builder.rs:379]: Note: A workaround to fix padding overflow
See https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1507

File: integration-tests/src/integration_test_circuits.rs [hunk 4]
```diff
@@ -131,13 +131,13 @@ lazy_static! {
...
+     /// Integration test for Exp circuit
+     pub static ref EXP_CIRCUIT_TEST: TokioMutex<IntegrationTest<TestExpCircuit::<Fr>>> =
+     TokioMutex::new(IntegrationTest::new("Exp", EXP_CIRCUIT_DEGREE, ROOT_CIRCUIT_SMALL_DEGREE));
 
-    /// Integration test for Pi circuit
-    pub static ref PI_CIRCUIT_TEST: TokioMutex<IntegrationTest<TestPiCircuit::<Fr>>> =
-    TokioMutex::new(IntegrationTest::new("Pi", PI_CIRCUIT_DEGREE, ROOT_CIRCUIT_SMALL_DEGREE));
+     /// Integration test for Pi circuit
+     pub static ref PI_CIRCUIT_TEST: TokioMutex<IntegrationTest<TestPiCircuit::<Fr>>> =
+     TokioMutex::new(IntegrationTest::new("Pi", PI_CIRCUIT_DEGREE, ROOT_CIRCUIT_SMALL_DEGREE));
```
Comment (ChihChengLiang) [inline integration-tests/src/integration_test_circuits.rs:140]: Nitpick: The formatting was breaking here.

**Review by ChihChengLiang — APPROVED** [review]
LGTM.
I added a nitpick.
I also highlighted some changes introduced only in this PR but not in previous PRs.

File: integration-tests/src/integration_test_circuits.rs (hunk 4 above)
Comment (hero78119) [reply integration-tests/src/integration_test_circuits.rs]: Thanks for catch up this. Fixed format problem in latest commit

File: zkevm-circuits/src/super_circuit/test.rs (hunk 1 above)
Comment (hero78119) [reply zkevm-circuits/src/super_circuit/test.rs:23]: A quick fix to reduce degree from 12 -> 10 in this pr https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1508. After pr is appoved and merged will update this pr as well

Comment (hero78119) [reply zkevm-circuits/src/super_circuit/test.rs]: Updated in latest commit. 

Final degree `9 -> 10`

**Pull request merged**